### PR TITLE
Retry jobs

### PR DESCRIPTION
**Pull request by nghi-ly**
## What are you changing in this pull request and why?

Add docs and release notes for Retry jobs:
- Add new Retry jobs page
- Update sidebar with new page under Deploy > Monitor jobs and alerts 
- Add release note 

## Checklist
- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Needs review from product team
- [ ] Needs this PR https://github.com/dbt-labs/cloud-ui/pull/2508 to merge

Adding new pages (delete if not applicable):
- [x] Add page to `website/sidebars.js`
- [x] Provide a unique filename for the new page

### CODE REVIEW

**Review by thorn14 — APPROVED** [review]
Looks good!

File: website/docs/docs/dbt-versions/release-notes/03-Oct-2023/native-retry-support-rn.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+title: "Enhancement: Native support for the dbt retry command"
+description: "October 2023: Rerun errored jobs from start and from the failure point"
```
Comment (runleonarun) [inline website/docs/docs/dbt-versions/release-notes/03-Oct-2023/native-retry-support-rn.md]: This can be either or, right? 
```suggestion
description: "October 2023: Rerun errored jobs from start or from the failure point"
```

File: website/docs/docs/deploy/retry-jobs.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+description: "Rerun your errored jobs from start or the failure point."
+---
+
+If your dbt job run completed with a status of `result:error` , you can rerun it from start or from the point of failure in dbt Cloud.
+
+## Prerequisites
+
+- You have a [dbt Cloud account](https://www.getdbt.com/signup).
+- You must be using [dbt version](/docs/dbt-versions/upgrade-core-in-cloud) 1.6 or newer.
+- There does not exist a more recent run of the job that completed successfully. The latest status of the run is `error`.
```
Comment (runleonarun) [inline website/docs/docs/deploy/retry-jobs.md]: Suggest a minor rewording to be more specific and active:
```suggestion
- The most recent run of the job has not completed successfully. The latest status of the run is `error`.
```

File: website/docs/docs/dbt-versions/release-notes/03-Oct-2023/native-retry-support-rn.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+sidebar_label: "Enhancement: Support for dbt retry"
+tags: [Oct-2023, Scheduler]
+date: 2023-10-06
+sidebar_position: 10
+---
+
+Previously in dbt Cloud, you could rerun an errored job from start but now you can rerun it from the job run's point of failure.
```
Comment (runleonarun) [inline website/docs/docs/dbt-versions/release-notes/03-Oct-2023/native-retry-support-rn.md]: Suggest calling out the two possible behaviors:
```suggestion
Previously in dbt Cloud, you could only rerun an errored job from start, but now you can either rerun a job from the start or from the job run's point of failure. 
```

File: website/docs/docs/deploy/retry-jobs.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    - The job command that failed in the run must be one that supports the [retry command](/reference/commands/retry).
+
+## Rerun an errored job
+
+1. Select **Deploy** from the top navigation bar and choose **Run History.** 
+2. Choose the job run that has errored. 
+3. In the **Run Summary** tab on the job’s **Run** page, expand the run step that failed. It’s denoted by an :x:.
```
Comment (runleonarun) [inline website/docs/docs/deploy/retry-jobs.md]: Suggest making the 2nd sentence active.
```suggestion
3. In the **Run Summary** tab on the job’s **Run** page, expand the run step that failed. An :x: denotes the failed step. 
```

File: website/docs/docs/deploy/retry-jobs.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+- There does not exist a more recent run of the job that completed successfully. The latest status of the run is `error`.
+    - The job command that failed in the run must be one that supports the [retry command](/reference/commands/retry).
+
+## Rerun an errored job
+
+1. Select **Deploy** from the top navigation bar and choose **Run History.** 
+2. Choose the job run that has errored. 
+3. In the **Run Summary** tab on the job’s **Run** page, expand the run step that failed. It’s denoted by an :x:. 
+4. Examine the error message and determine how to fix it. After you have made your changes, save and commit them to your [Git repo](/docs/collaborate/git-version-control).
+5. Go back to your job’s **Run** page. In the upper right corner, click **Rerun** and choose **Rerun from start** or **Rerun from failure**.
```
Comment (runleonarun) [inline website/docs/docs/deploy/retry-jobs.md]: Are they clicking back or just returning? Suggest being explicit to help those using screen readers. 
```suggestion
5. Return to your job’s **Run** page. In the upper right corner, click **Rerun** and choose **Rerun from start** or **Rerun from failure**.
```

File: website/docs/docs/deploy/retry-jobs.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    
+    If you choose to rerun from the failure point, a **Rerun failed steps** modal will appear. It lists the run steps it will invoke which will include the failed step and the steps it previously wasn’t able to invoke (skipped steps). To confirm, click **Rerun from failure**. The job reruns from the failed command in the previously failed run. This is denoted at the top of the **Run Summary** tab by a message "This run resumed execution from last failed step".
```
Comment (runleonarun) [inline website/docs/docs/deploy/retry-jobs.md]: I got lost in the "its." Suggest rewording to clarify and tighten the sentence. 
Should this be a  new step?
```suggestion
    If you chose to rerun from the failure point, a **Rerun failed steps** modal opens showing the run steps this process will invoke. Proceeding reruns the failed step as well as any skipped steps. To confirm, click **Rerun from failure**. The job reruns from the failed command in the previously failed run. The **Run Summary** tab captures the status in the message, "This run resumed execution from last failed step."
```

**Review by runleonarun — COMMENTED** [review]
Hey @nghi-ly! I just have a few suggestions to tighten up a couple of sentences. 

I like the use of "modal" here and feel like I've also seen "pop up," but I think we should add modal to our style guide.

File: website/docs/docs/dbt-versions/release-notes/03-Oct-2023/native-retry-support-rn.md (hunk 1 above)
Comment (nghi-ly) [reply website/docs/docs/dbt-versions/release-notes/03-Oct-2023/native-retry-support-rn.md:3]: not in this case, i don't think. it's more similar to this:

"foo supports x, y, and z"

Comment (nghi-ly) [reply website/docs/docs/dbt-versions/release-notes/03-Oct-2023/native-retry-support-rn.md]: weird i had it on the page "Retry jobs"

File: website/docs/docs/deploy/retry-jobs.md (hunk 6 above)
Comment (nghi-ly) [reply website/docs/docs/deploy/retry-jobs.md]: good flag! i see what you mean. lemme make this a bit more clear and concise. thanks!

File: website/docs/docs/dbt-versions/release-notes/03-Oct-2023/native-retry-support-rn.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+---
+title: "Enhancement: Native support for the dbt retry command"
+description: "October 2023: Rerun errored jobs from start or from the failure point"
+sidebar_label: "Enhancement: Support for dbt retry"
+tags: [Oct-2023, Scheduler]
+date: 2023-10-06
+sidebar_position: 10
+---
+
+Previously in dbt Cloud, you could only rerun an errored job from start but now you can also rerun it from its of failure.
```
Comment (nghi-ly) [inline website/docs/docs/dbt-versions/release-notes/03-Oct-2023/native-retry-support-rn.md]:
```suggestion
Previously in dbt Cloud, you could only rerun an errored job from start but now you can also rerun it from its point of failure. 
```

File: website/docs/docs/deploy/retry-jobs.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+## Rerun an errored job
+
+1. Select **Deploy** from the top navigation bar and choose **Run History.** 
+2. Choose the job run that has errored. 
+3. In the **Run Summary** tab on the job’s **Run** page, expand the run step that failed. An :x: denotes the failed step. 
+4. Examine the error message and determine how to fix it. After you have made your changes, save and commit them to your [Git repo](/docs/collaborate/git-version-control).
+5. Return to your job’s **Run** page. In the upper right corner, click **Rerun** and choose **Rerun from start** or **Rerun from failure**.
+    
+    If you chose to rerun from the failure point, a **Rerun failed steps** modal opens. It lists the run steps it will invoke: the failed step and any skipped steps. To confirm these run steps, click **Rerun from failure**. The job reruns from the failed command in the previously failed run. A banner at the top of the **Run Summary** tab captures this with the message, "This run resumed execution from last failed step".
```
Comment (runleonarun) [inline website/docs/docs/deploy/retry-jobs.md]: > It lists the run steps it will invoke:

Possibly confusing for non-English speakers only because the 1st "it" refers to the modal and the second "it" refers to dbt.

For the last sentence could use a little clarity in what the user should do. Should they check the banner for xyz success message? Is there a message they should worry about?

**Review by runleonarun — APPROVED** [review]
One more non-blocking comment/question. Thanks @nghi-ly! This page will be super helpful!

File: website/docs/docs/deploy/retry-jobs.md (hunk 8 above)
Comment (nghi-ly) [reply website/docs/docs/deploy/retry-jobs.md]: - i am referring to the first modal here so it is the first "it". i was taught this was a valid use of "it". you're right about the second "it". will clarify, good catch!
- nothing the user should do which is why it's phrased the way it is, just describing the banner and pointing the banner out to the user
